library/Makefile

```diff
@@ -5,7 +5,20 @@ CFLAGS	?= -O2
 WARNING_CFLAGS ?=  -Wall -W -Wdeclaration-after-statement
 LDFLAGS ?=
 
-LOCAL_CFLAGS = $(WARNING_CFLAGS) -I../include -D_FILE_OFFSET_BITS=64
+DIR_FOR_MBED_TLS_ENV=../library
+ifneq "$(wildcard $(DIR_FOR_MBED_TLS_ENV) )" ""
+    # Set include dirs for Mbed TLS test environment
+	INCLUDE_DIRS=-I../include
+else
+    # Set include dirs for Mbed OS test environment
+	INCLUDE_DIRS=-I.. -I../inc
+endif
+
+LOCAL_CFLAGS = $(WARNING_CFLAGS) $(INCLUDE_DIRS) -D_FILE_OFFSET_BITS=64
+ifdef MBEDTLS_CONFIG_FILE
+LOCAL_CFLAGS += "-DMBEDTLS_CONFIG_FILE=\"${MBEDTLS_CONFIG_FILE}\""
```

Comment:
Non-blocking: I'd prefer to stick with manual uses of `make CFLAGS='-DMBEDTLS_CONFIG_FILE=\"...\"'` than using an environment variable of the same name here. Otherwise, there might be expectation that the _environment variable_ `MBEDTLS_CONFIG_FILE` can also be used in CMake builds, which isn't the case.